.github/workflows/ci.yaml

```diff
@@ -133,6 +133,54 @@ jobs:
         if: matrix.os == 'ubuntu-20.04' || matrix.os == 'macos-latest'
         run: ./tools/ensure_todoapp_builds.sh
 
+      - name: Headless - Cache Node Modules
+        id: headless-cache-node-modules
+        uses: actions/cache@v3
+        if: matrix.os == 'ubuntu-20.04'
+        with:
+          path: ~/.npm
+          key: node-modules-${{ hashFiles('./waspc/headless-test/package-lock.json') }}
+
+      - name: Headless - Install Dependencies
+        id: headless-install-dependencies
+        if: matrix.os == 'ubuntu-20.04'
+        run: |
+          cd headless-test
+          npm ci
+
+      - name: Headless - Store Playwright's Version
+        id: headless-store-playwright-version
+        if: matrix.os == 'ubuntu-20.04'
```

Comment:
I did try to solve the issue of grouping similar steps, it annoyed me that I had to repeat conditions, prefix the names etc. 

The issue with defining a new job is: Github Actions use a new isolated runner for each job - meaning we would need to set up Haskell, build Wasp etc. in the new job which I wanted to avoid.

Reply:
Ah ok, I was hoping there is an easy way to have them reuse the runner / container, but it seems like there is not. Ok then, what can we do! Maaaaybe we could use yaml feature, I think they have some way to define variables or something? So we at least have one single place where we need to update that ubuntu version if we want to? Maybe that is worth giving a quick look.